### PR TITLE
[Author][New] Add CreateJS@1.0.1 w/ git auto-update

### PR DESCRIPTION
**Pull request by tehvgg**
cc @wdamien 

**NOTE**: Popularity numbers are calculated as the combined values from the 4 libraries inside CreateJS (EaselJS, TweenJS, SoundJS, PreloadJS)

Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/CreateJS/CreateJS
 * Official website (optional, not the repository): https://createjs.com
 * NPM package url (optional):
 * License and its reference: MIT https://github.com/CreateJS/CreateJS/blob/master/LICENSE
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 921
   - Count of stars: 14,224
   - Count of forks: 3,604
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [X] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [X] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [X] More than 800 downloads stats per month on npm registry
 * [X] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [X] Has valid tags for each versions (for git auto-update)
 * [X] Auto-update setup
 * [X] Auto-update target/source is valid.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
@tehvgg Thanks for the PR, however, this library current has 8 stars on GutHub repo and doesn't have enough notable popularity(like more than 200 stars on GitHub or 800 downloads per month on npm). please take a look on our [README.md](https://github.com/cdnjs/cdnjs/blob/master/README.md) for detail, it's always good to read the document before open an issue, I'll respectfully suggest you send a new PR again when the library has some notable popularity.

To prevent I missed anything (e.g. It's a famous library, but only publish on GitHub or npm recently so it doesn't have a notable popularity on these platforms, or maybe it's a js library developed for a popular application), I will not close this issue in next 7 days, feel free to show some other references if you have something else want to discuss with us, if there is no any response in next 7 day, this issue will be closed.

Thanks for being interested in CDNJS.

**Comment by tehvgg** [comment]
Hi @sufuf3, I'll clarify my note at the top of the PR.

CreateJS is a suite of 4 libraries, all of which are already hosted on CDNJS. For convenience, we also build a combined version of the libraries. This is that combined build.

EaselJS
https://github.com/createjs/easeljs
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/EaselJS

TweenJS
https://github.com/createjs/tweenjs
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/TweenJS

SoundJS
https://github.com/createjs/soundjs
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/SoundJS

PreloadJS
https://github.com/createjs/preloadjs
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/PreloadJS

The combined version is normally hosted here (https://github.com/createjs/combined) but we are moving it to this new repository as the first of many changes to come for our repository structure.

The star/watch/fork counts I provided in the PR are the combined numbers for those 4 libraries.

**Comment by PeterDaveHello** [comment]
@tehvgg Thanks

**Comment by sebastianderossi** [comment]
Hi @PeterDaveHello,  just wondering what that is status on this issue, has there been any progress on this?  Thanks!

**Comment by PeterDaveHello** [comment]
@sebastianderossi hmmmm, we'll take care of it soon :)

**Comment by sebastianderossi** [comment]
@PeterDaveHello Perfect! Thanks 🥇:D

File: ajax/libs/CreateJS/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  },
+  "license": "MIT",
+  "author": "gskinner.com, inc.",
+  "homepage": "https://createjs.com/",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/CreateJS/CreateJS.git",
+    "fileMap": [
+      {
+        "basePath": "dist/",
```
Comment (extend1994) [inline ajax/libs/CreateJS/package.json]: Let's use `dist`.

File: ajax/libs/CreateJS/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "files": [
+          "createjs.js",
```
Comment (extend1994) [inline ajax/libs/CreateJS/package.json]: Let's just use `*/**` to fetch all files under `dist`.

**Comment by tehvgg** [comment]
Rebased my branch with master to resolve CI errors about being out-of-date.

File: ajax/libs/CreateJS/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "author": "gskinner.com, inc.",
+  "homepage": "https://createjs.com/",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/CreateJS/CreateJS.git",
+    "fileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "*/**"
```
Comment (extend1994) [inline ajax/libs/CreateJS/package.json]: I am so sorry about the wrong request I made! 😞 This should be `**/*`.

**Comment by tehvgg** [comment]
v1.0.1 is taken from the latest release found here https://github.com/CreateJS/Combined/releases/tag/v1.0.1 which is the old repository that we are in the process of deprecating.

Fixed the file glob.

**Comment by tehvgg** [comment]
@sufuf3 

I'm not sure which files you're comparing, but the build files found in the zip here https://github.com/CreateJS/CreateJS/releases/tag/v1.0.1 and the files in this PR are identical. I deleted the ones in the PR and copied over the build files and git registered no changes.

I've amended the commit message to your desired format.

**Comment by tehvgg** [comment]
Rebased with upstream master.

**Comment by extend1994** [comment]
Sorry for late review.
@tehvgg sufuf3 was comparing the file in your PR and two files under https://github.com/CreateJS/CreateJS/tree/v1.0.1/dist
Unfortunately, I still get different file content from yours (mine is from the above link). There are lots of files under `build` in v1.0.1 you mentioned. Maybe you can try to re-add file from https://github.com/CreateJS/CreateJS and let's see if it's the same as the ones we get. I think the future release will be at https://github.com/CreateJS/CreateJS?

**Comment by tehvgg** [comment]
Here are the files in this PR.
https://raw.githubusercontent.com/tehvgg/cdnjs/216b0126d1d7990d49dff348eb0cec03910cf24f/ajax/libs/CreateJS/1.0.1/createjs.js
https://raw.githubusercontent.com/tehvgg/cdnjs/216b0126d1d7990d49dff348eb0cec03910cf24f/ajax/libs/CreateJS/1.0.1/createjs.min.js
Here are the files in the repository.
https://raw.githubusercontent.com/CreateJS/CreateJS/v1.0.1/dist/createjs.js
https://raw.githubusercontent.com/CreateJS/CreateJS/v1.0.1/dist/createjs.min.js

Our build process injects the build date into the file. If you scroll to the bottom of the file, you'll see that they all say exactly "Thu, 14 Sep 2017 22:19:45 GMT"

Like I said in my last comment, I did update the files by deleting the ones in the PR and re-adding from the repository, and Git didn't register any changes.

I've run SHA1 sums on the files (on windows), here are the results.
```
CertUtil -hashfile PR-createjs.js SHA1 -> 1a52e870794a1a72811962ad4bcb22ccb27f8195
CertUtil -hashfile RE-createjs.js SHA1 -> 49b322bed4dfb5cafa468bb6cb854f2421d20298

CertUtil -hashfile PR-createjs.min.js SHA1 -> f0fa1d43a40b08489ef119c41174bbae16781367
CertUtil -hashfile RE-createjs.min.js SHA1 -> f0fa1d43a40b08489ef119c41174bbae16781367
```
The unminified files have different hashes, but Git doesn't see a difference. They are identical line length and build date. I've uploaded them to this online merge tool to illustrate what git's diff checker is seeing here http://www.mergely.com/4Wyy4dpP/

As the person who committed the files to https://github.com/createjs/createjs and published the v1.0.1 release there, I'm confirming that the files are identical.

**Comment by extend1994** [comment]
@tehvgg I think it's related to this! https://help.github.com/articles/dealing-with-line-endings/#platform-all
Would you like to try to set it up?
(@sufuf3 and I both use Linux systems)

**Comment by extend1994** [comment]
@tehvgg Our maintainer asked us to check if the checksum of the files from upstream  is the same as the files in the PR to make sure file integrity is fine. 
@PeterDaveHello Could you please help review this? It's the same issue in  #12042. Thank you!

**Comment by PeterDaveHello** [comment]
@extend1994 sure, I'll take a look.

**Comment by sebastianderossi** [comment]
Hi @PeterDaveHello,
Just wanted to check the status of this. Has there been any progress on this? 
Thanks!

**Comment by PeterDaveHello** [comment]
@sebastianderossi will check soon, there are too many pull requests lol ...

**Comment by sebastianderossi** [comment]
That's a good question @PeterBot ;) @PeterDaveHello

**Comment by PeterDaveHello** [comment]
@sebastianderossi My bad . . .

**Comment by sebastianderossi** [comment]
@PeterDaveHello Ping :)

**Comment by sashberd** [comment]
@MattIPv4 Should we continue with this one?

**Comment by MattIPv4** [comment]
Hm, actually, looking at this it would seem only one version was ever released here. I don't think this is worth continuing with as we already have all the separate libraries hosted.

**Pull request closed without merging**